modules/openxBidAdapter.js

```diff
@@ -26,35 +27,66 @@ export const spec = {
       return;
     }
 
-    let delDomain = bids[0].params.delDomain;
-    let configuredBc = bids[0].params.bc;
-    let bc = configuredBc || `${BIDDER_CONFIG}_${BIDDER_VERSION}`;
-
-    return buildOXRequest(bids, {
-      ju: currentURL,
-      jr: currentURL,
-      ch: document.charSet || document.characterSet,
-      res: `${screen.width}x${screen.height}x${screen.colorDepth}`,
-      ifr: isIfr,
-      tz: new Date().getTimezoneOffset(),
-      tws: getViewportDimensions(isIfr),
-      ef: 'bt%2Cdb',
-      be: 1,
-      bc: bc,
-      nocache: new Date().getTime()
-    },
-    delDomain);
+    let requests = [];
+    let banner_requests = [];
+    let video_requests = [];
+    let bids_banner = bids.filter(function(bid) { return bid.mediaType !== 'video'; });
+    let bids_video = bids.filter(function(bid) { return bid.mediaType === 'video'; });
+
+    // build banner requests
+    if (bids_banner.length !== 0) {
+      let delDomain = bids_banner[0].params.delDomain;
+      let configuredBc = bids_banner[0].params.bc;
+      let bc = configuredBc || `${BIDDER_CONFIG}_${BIDDER_VERSION}`;
+      banner_requests = [ buildOXRequest(bids_banner, {
+        ju: currentURL,
+        jr: currentURL,
+        ch: document.charSet || document.characterSet,
+        res: `${screen.width}x${screen.height}x${screen.colorDepth}`,
+        ifr: isIfr,
+        tz: new Date().getTimezoneOffset(),
+        tws: getViewportDimensions(isIfr),
+        ef: 'bt%2Cdb',
+        be: 1,
+        bc: bc,
+        nocache: new Date().getTime()
+      },
+      delDomain)];
+    }
+
+    // build video requests
+    video_requests = buildOXVideoRequest(bids_video);
+    requests = banner_requests.concat(video_requests);
+    // return requests;
+    return requests;
   },
   interpretResponse: function(oxResponseObj, bidRequest) {
     let bidResponses = [];
-    let adUnits = oxResponseObj.ads.ad;
-    if (oxResponseObj.ads && oxResponseObj.ads.pixels) {
-      userSync.registerSync('iframe', 'openx', oxResponseObj.ads.pixels);
+    let mediaType = 'banner';
+    if (bidRequest && bidRequest.payload) {
+      if (bidRequest.payload.bids) {
+        mediaType = bidRequest.payload.bids[0].mediaType;
+      } else if (bidRequest.payload.bid) {
+        mediaType = bidRequest.payload.bid.mediaType;
+      }
     }
-    if (!adUnits) {
-      adUnits = [];
+    if (mediaType === 'video') {
+      const parts = oxResponseObj.match(/^[a-z0-9]+\(([^\)]+)\)/i);
+      let response = !parts || parts.length != 2 ? undefined : JSON.parse(parts[1]);
+      if (response && response.pixels) {
+        userSync.registerSync('iframe', 'openx', response.pixels);
+      }
+      bidResponses = createVideoBidResponses(response, bidRequest.payload);
```

Comment:
could possibly exit early and return bidResponses here, avoiding an 'else' that might lead into trouble... (e.g. adding a media type leads into a problem for whoever needs to add that media type)